src/borg/helpers/fs.py

```diff
@@ -12,18 +12,22 @@
 from ..constants import *  # NOQA
 
 
-def get_home_dir():
-    """Get user's home directory while preferring a possibly set HOME
-    environment variable
+def get_base_dir():
+    """Get home directory / base directory for borg:
+
+    - BORG_BASE_DIR, if set
+    - HOME, if set
+    - ~$USER, if USER is set
+    - ~
     """
+    base_dir = os.environ.get('BORG_BASE_DIR') or os.environ.get('HOME')
     # os.path.expanduser() behaves differently for '~' and '~someuser' as
     # parameters: when called with an explicit username, the possibly set
     # environment variable HOME is no longer respected. So we have to check if
     # it is set and only expand the user's home directory if HOME is unset.
-    if os.environ.get('HOME', ''):
-        return os.environ.get('HOME')
-    else:
-        return os.path.expanduser('~%s' % os.environ.get('USER', ''))
+    if not base_dir:
+        base_dir = os.path.expanduser('~%s' % os.environ.get('USER', ''))
+    return base_dir
```

Comment:
Might be nice to have some test coverage for this function, to go with the other `get_*_dir()` tests: https://github.com/borgbackup/borg/blob/master/src/borg/testsuite/helpers.py#L20 .. You could assert that given a particular `os.environ` dict, the returned base dir is as expected.

Reply:
done.